voice/src/main/kotlin/encryption/VoiceEncryption.kt

```diff
@@ -0,0 +1,163 @@
+package dev.kord.voice.encryption
+
+import com.iwebpp.crypto.TweetNaclFast
+import dev.kord.voice.EncryptionMode
+import dev.kord.voice.encryption.strategies.LiteNonceStrategy
+import dev.kord.voice.encryption.strategies.NonceStrategy
+import dev.kord.voice.io.ByteArrayView
+import dev.kord.voice.io.MutableByteArrayCursor
+import dev.kord.voice.io.mutableCursor
+import dev.kord.voice.io.view
+import dev.kord.voice.udp.RTPPacket
+import javax.crypto.Cipher
+import javax.crypto.spec.GCMParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+public sealed interface VoiceEncryption {
+    public val mode: EncryptionMode
+
+    public val nonceLength: Int
+
+    public val supportsDecryption: Boolean get() = true
```

Comment:
not having a default implementation here will avoid mistakes if another `VoiceEncryption` is added at some point